lib/room.cpp

```diff
@@ -1529,8 +1506,23 @@ void Room::handleRoomKeyEvent(const RoomKeyEvent& roomKeyEvent,
     }
     if (d->addInboundGroupSession(senderKey, roomKeyEvent.sessionId(),
                                   roomKeyEvent.sessionKey())) {
-        qCDebug(E2EE) << "added new inboundGroupSession:"
-                      << d->groupSessions.count();
+        qCWarning(E2EE) << "added new inboundGroupSession:"
+                      << d->groupSessions.size();
+        for (const auto& eventId : d->undecryptedEvents[roomKeyEvent.sessionId()]) {
+            if (!d->eventsIndex.contains(eventId)) {
+                continue;
+            }
+            auto event = d->timeline.rend() - (d->eventsIndex.value(eventId) - minTimelineIndex() + 1);
+            if (auto encryptedEvent = event->viewAs<EncryptedEvent>()) {
+                auto decrypted = decryptMessage(*encryptedEvent);
+                if(decrypted) {
+                    auto oldEvent = event->replaceEvent(std::move(decrypted));
+                    decrypted->setOriginalEvent(std::move(oldEvent));
+                    emit replacedEvent(event->event(), rawPtr(oldEvent));
```

Comment:
Beware of using pointers after `std::move` - your code here fails twice in that.
```suggestion
                    // The reference will survive the pointer being moved
                    auto& decryptedEvent = *decrypted;
                    auto oldEvent = ti.replaceEvent(std::move(decrypted));
                    decryptedEvent.setOriginalEvent(std::move(oldEvent));
                    emit replacedEvent(ti.event(), decrypted->originalEvent());
```